### ISSUE TITLE
Pretty affiliate links are not working with custom category permalinks

### ISSUE THREAD

**Issue by michaelbeil** [AffiliateWP/theme#143]
> Hi team. Hope all is well. Not sure what is going on and/or if I am missing something obvious, but currently my affiliate link created by the link generator int he affiliate area is resulting in a 404.
> 
> I want to link to this page here - https://affiliatewp.com/add-ons/official-free/
> 
> So I put that int he generator with a campaign ID of deals-page and get this URL:
> 
> https://affiliatewp.com/add-ons/official-free/ref/305/?campaign=deals-page
> 
> Which results in a 404. If I change it to the other format like this:
> 
> https://affiliatewp.com/add-ons/official-free/?ref=305&campaign=deals-page
> 
> Also did not work without the campaign ID:
> 
> https://affiliatewp.com/add-ons/official-free/ref/305/
> 
> That seems to work. Hope that helps and all the best.

Ref: https://secure.helpscout.net/conversation/276325498/46572/?folderId=205524

**Comment by sumobi** [comment]
@pippinsplugins 

I think I might remove some code in the theme that adds the taxonomy to the URL. The code allows us to have `pro` (the download category) in this URL: 

https://affiliatewp.com/add-ons/pro/rest-api-extended/

Although pretty affiliate links can be used on the above URL, it doesn't work for the following URLs and show a 404 page not found:

https://affiliatewp.com/add-ons/ref/sumobi
https://affiliatewp.com/add-ons/pro/ref/sumobi

We currently don't show pretty affiliate links to affiliates (maybe in the future?) but since we sell affiliate marketing software I think it's only right that **all** our URLs work.

Here's what our URLs look like currently versus what they will look like after. We can simply set up 301 redirects.

### Current URLs:

Download category page:
https://affiliatewp.com/add-ons/pro/

Single add-on page:
https://affiliatewp.com/add-ons/pro/rest-api-extended/



### New URLs:

Download category page: (has `category` in the URL)
https://affiliatewp.com/add-ons/category/pro/

Single add-on page: (no longer has `official-free` or `pro` in the URL
https://affiliatewp.com/add-ons/rest-api-extended/

**Comment by pippinsplugins** [comment]
I agree, let's remove it, at least until such time as ref links work on it.

**Comment by sumobi** [comment]
👍

**Comment by sumobi** [comment]
I'm going to make a list of redirects for @cklosowski and at the same time as I push an update for this I'll get him to add them.
